runtime/src/bank.rs

```diff
@@ -1627,19 +1628,27 @@ impl Bank {
                     self.get_account(&delegation.voter_pubkey),
                 ) {
                     (Some(stake_account), Some(vote_account)) => {
-                        let vote_owner = vote_account.owner;
-                        let entry = accounts
-                            .entry(delegation.voter_pubkey)
-                            .or_insert((Vec::new(), vote_account));
+                        // call tracer to catch any illegal data if any
                         if let Some(reward_calc_tracer) = reward_calc_tracer {
                             reward_calc_tracer(&RewardCalculationEvent::Staking(
                                 stake_pubkey,
                                 &InflationPointCalculationEvent::Delegation(
                                     *delegation,
-                                    vote_owner,
+                                    vote_account.owner,
                                 ),
                             ));
                         }
+                        if self
+                            .feature_set
+                            .is_active(&feature_set::filter_stake_delegation_accounts::id())
+                            && (stake_account.owner != solana_stake_program::id()
+                                || vote_account.owner != solana_vote_program::id())
+                        {
```

Comment:
@ryoqun what reporting do you think would be most helpful here?

Reply:
hmm, I think just `error!()` suffices. At least this is more significant than `warn!`, imo.

Reply:
That seems a bit strong to me; it's not really a node or runtime error, is it? Would a metric datapoint reporting the slot and the address perhaps be more useful?

Reply:
> That seems a bit strong to me; it's not really a node or runtime error, is it? Would a metric datapoint reporting the slot and the address perhaps be more useful?

Hhhm, yeah, I agree. After more thought, I think we can go with your suggested approach.